modules/integrations/arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/syntax.kt

```diff
@@ -0,0 +1,36 @@
+package arrow.integrations.retrofit.adapter
+
+import arrow.Kind
+import arrow.effects.typeclasses.Async
+import arrow.effects.typeclasses.MonadDefer
+import arrow.typeclasses.ApplicativeError
+import arrow.typeclasses.MonadError
+import retrofit2.Call
+import retrofit2.HttpException
+import retrofit2.Response
+
+fun <F, R> Response<R>.unwrapBody(apError: ApplicativeError<F, Throwable>): Kind<F, R> =
+  if (this.isSuccessful) {
+    val body = this.body()
+    if (body != null) {
+      apError.just(body)
+    } else {
+      apError.raiseError(IllegalStateException("The request returned a null body"))
+    }
+  } else {
+    apError.raiseError(HttpException(this))
+  }
+
+fun <F, A> Call<A>.runInAsyncContext(AC: Async<F>): Kind<F, Response<A>> =
```

Comment:
`runAsync`? I believe this name is a holdover from when `Async` was called `AsyncContext`. Have you seen it somewhere else?

Reply:
I just didn't know any good name haha. `runAsync` looks good